py/vm.c

```diff
@@ -148,7 +208,46 @@ mp_vm_return_kind_t mp_execute_bytecode(mp_code_state_t *code_state, volatile mp
 
 #if MICROPY_STACKLESS
 run_code_state: ;
+#endif // MICROPY_STACKLESS
+FRAME_ENTER();
+
+#if MICROPY_PY_SYS_TRACE
+// Trace CALL event.
+if (!prof_is_executing && prof_trace_cb) {
+
```

Comment:
The following chunk of code can probably be factored to a function in `profiling.c`, then `prof_callback_invoke()` can be made private to `profiling.c`.